src/amo/reducers/addonsByAuthors.js

```diff
@@ -115,12 +174,10 @@ const reducer = (
         };
       }
 
-      // Reset the data for each author requested.
```

Comment:
Because resetting the data means you can't have multiple requests/reducer datasets for the same author on a page. In the current (pre-this-patch) implementation of this reducer, add-ons for an author were reset whenever the fetch action was dispatched. But if we want to show themes and extensions for a given user on the page we need to have the ability to add add-ons to the author's list without resetting it first. We'll still make new fetch requests when the components load so we won't "cache" stale data, but this way we can have multiple responses on a page.

The old reducer was a bit too naïve about its use-case, this needs to be removed to allow a UserProfile page with multiple cards.